### ISSUE TITLE
Rewrite screenshot script in python3 and Gtk3

### ISSUE THREAD

**Issue by ftsamis** [Epoptes/epoptes#46]
The screenshot script is executed on the clients in order to receive a single screenshot from them with certain dimensions (width, height), specified as arguments to the script. The script's output is printed on stdout, along with the effective width, height and rowstride (which may not be necessary). Epoptes then reads this output from the client and converts it back to a pixbuf to display in the application.

This is currently written in Python2 + pygtk. It should be ported in Python3 + Gtk3.

We have to make sure that the script, and mainly the screenshot's scaling to the desired resolution is as efficient/lightweight (resource-wise) as possible. Some benchmarks of different implementations could be a good way to assure that.

**Comment by cshubhamrao** [comment]
Ported the screenshot script to Python3 + PyGObject. Please have a look here: [(gist) cshubhamrao/screenshot3](https://gist.github.com/cshubhamrao/76db45d56bae360d882329d8f4abb80e)

I just have one doubt: How to approach the benchmarking part? What to parameters to look out for?

**Comment by ftsamis** [comment]
Have you tried running Epoptes with a client using this script? Do you see
the thumbnails properly?

Also please make this a PR instead of a gist.

2018-03-15 9:32 GMT+02:00 Shubham Rao <notifications@github.com>:

**Comment by alkisg** [comment]
I did an initial benchmark. I cannot reproduce the cairo-based speed up that we had observed in thin clients anymore.
Specifically, I booted an Ubuntu 18.04/LTSP thin client over a deliberately slow 10 Mbps link, I logged in, and I ran
`time /usr/share/epoptes-client/screenshot 10 10 >/dev/null`

It needed about 4 seconds in both the original and the updated versions, which means the stretching is happening client-side instead of server-side, as we were hoping.
I then tried to disable compositing by running
`marco --no-composite --replace`
...to no avail, the results were the same.

On the contrary, running the screenshot command as root locally on the thin client, only needed 0.25 seconds, so maybe we can just change the server-side `execOnClients('screenshot'...)` call to EM_SYSTEM_OR_SESSION, while making sure it doesn't run before login, to save bandwidth.

Chatting about the issue in #cairo, I got the following input; I note it here in case it proves useful later on:

> (02:40:01 μμ) The_Company: alkisg: you want to use https://www.cairographics.org/manual/cairo-XLib-Surfaces.html#cairo-xlib-surface-create for the root window, but I have no idea how to do that in Python
(02:41:31 μμ) The_Company: and instead of the pixmap, you want to https://www.cairographics.org/manual/cairo-cairo-surface-t.html#cairo-surface-create-similar
(02:44:19 μμ) The_Company: and finally, you probably want to use https://www.cairographics.org/manual/cairo-cairo-surface-t.html#cairo-surface-map-to-image and https://www.cairographics.org/manual/cairo-Image-Surfaces.html#cairo-image-surface-get-data instead of GdkPixbuf
(02:44:28 μμ) The_Company: even though that uses a different pixel format
